sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -948,6 +975,14 @@ public void setTimestampPolicy(String timestampPolicy) {
         public void setConsumerPollingTimeout(Long consumerPollingTimeout) {
           this.consumerPollingTimeout = consumerPollingTimeout;
         }
+
+        public void setRedistributeNumShards(Long redistributeNumShards) {
+          this.redistributeNumShards = redistributeNumShards;
+        }
+
+        public void setRedistribute(Boolean redistribute) {
```

Comment:
You also need to have `allowDuplicates` as a parameter. It makes sense to also Redistribute for exactly once use case.

Reply:
Done.